### PR TITLE
Port SBOM generation to v3.x/ps6

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-powershell-worker/issues/721

Cherry-picked https://github.com/Azure/azure-functions-powershell-worker/pull/716, https://github.com/Azure/azure-functions-powershell-worker/pull/718, https://github.com/Azure/azure-functions-powershell-worker/pull/720, https://github.com/Azure/azure-functions-powershell-worker/pull/728, https://github.com/Azure/azure-functions-powershell-worker/pull/731 applicable changes to the `v3.x/ps6` branch.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: azure-pipelines.yml
Comment (davidmrdavid) [inline azure-pipelines.yml:34]: Should we hardcode the commit of this binary? Otherwise, we risk changing it changing, and breaking, silently

Comment (davidmrdavid) [inline azure-pipelines.yml:34]: Should we hardcode the commit of this binary? Otherwise, we risk changing it changing, and breaking, silently

**Review by davidmrdavid — APPROVED** [review]
Approved but I'd prefer if we could _set_ the specific commit of the `dotnet-install.ps1` resource that we're downloading. However, I also understand that this is a trusted resource, so I'll leave it up to you to make that change if you think it's worth it.

Also, before merge, let's make sure the CI tests pass :)

File: azure-pipelines.yml
Comment (Francisco-Gamino) [reply azure-pipelines.yml:29]: Good point. However, we have not had any issues on this regard. We use this same script to install .Net in other repros when building, this includes the the Core Tools and the .Net worker. Here is a link: 
https://github.com/Azure/azure-functions-dotnet-worker/blob/main/build/install-dotnet.yml

Comment (Francisco-Gamino) [reply azure-pipelines.yml:34]: Good point. However, we have not had any issues on this regard. We use this same script to install .Net in other repros when building, this includes the the Core Tools and the .Net worker. Here is a link:
https://github.com/Azure/azure-functions-dotnet-worker/blob/main/build/install-dotnet.yml

**Review by michaelpeng36 — APPROVED** [review]
LGTM

**Pull request merged**
